### PR TITLE
StreamAllocator (congestion controller) refactor

### PR DESCRIPTION
**Pull request by boks1971**
No functional change, just refactoring `StreamAllocator (congestion controller)`.

- BWE interface. `RemoteBWE` is the existing method refactored out of `StreamAllocator`. `SendSideBWE` is the TWCC based one, this has just moved directories
- Probing is still not clean. I am not happy with it. But, making it so that it works for existing case. Will take another look to make it cleaner when implementing probing for TWCC (this is still pending). But, wanted to get in this as it is already big and make more progress after this is merged (also reduces cognitive load for my head :-) )

### CODE REVIEW

File: pkg/sfu/bwe/remotebwe/remote_bwe.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+	commitThreshold := int64(r.params.Config.ExpectedUsageThreshold * float64(r.lastExpectedBandwidthUsage))
+	action := "applying"
+	if estimateToCommit > commitThreshold {
+		action = "skipping"
+	}
+
+	if action == "applying" {
+		r.params.Logger.Infow(
+			fmt.Sprintf("remote bwe: channel congestion detected, %s channel capacity update", action),
```
Comment (paulwe) [inline pkg/sfu/bwe/remotebwe/remote_bwe.go:205]: these sprintfs are only ever called with 1 unique value each?

File: pkg/sfu/bwe/remotebwe/remote_bwe.go [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+			"new(bps)", estimateToCommit,
+			"lastReceived(bps)", r.lastReceivedEstimate,
+			"expectedUsage(bps)", r.lastExpectedBandwidthUsage,
+			"commitThreshold(bps)", commitThreshold,
+			"channel", r.channelObserver.ToString(),
+		)
+	}
+	/*
+		r.params.Logger.Debugw(
+			fmt.Sprintf("remote bwe: channel congestion detected, %s channel capacity: experimental", action),
```
Comment (paulwe) [inline pkg/sfu/bwe/remotebwe/remote_bwe.go:228]: duplicated above?

File: pkg/sfu/bwe/remotebwe/remote_bwe.go [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+func (r *RemoteBWE) ProbingStart(expectedBandwidthUsage int64) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	r.isInProbe = true
+	r.lastExpectedBandwidthUsage = expectedBandwidthUsage
+
+	channelState := ""
+	if r.channelObserver != nil {
+		channelState = r.channelObserver.ToString()
```
Comment (paulwe) [inline pkg/sfu/bwe/remotebwe/remote_bwe.go:283]: `stringer` interface?

**Review by paulwe — APPROVED** [review]

File: pkg/sfu/bwe/remotebwe/remote_bwe.go (hunk 3 above)
Comment (boks1971) [reply pkg/sfu/bwe/remotebwe/remote_bwe.go:283]: Will change these. Sorry, I picked up that habit from somewhere when I started using Go and that has just stuck. Don't remember where I saw it and picked up.

File: pkg/sfu/bwe/remotebwe/remote_bwe.go (hunk 1 above)
Comment (boks1971) [reply pkg/sfu/bwe/remotebwe/remote_bwe.go:205]: Will make an unlikely logger. Yeah, the difference is mainly `Infow` vs `Debugw`.

**Pull request merged**